config/config.go

```diff
@@ -100,32 +89,41 @@ func Parse(cfg map[string]string) (Config, error) {
 		return Config{}, fmt.Errorf("invalid polling duration passed(%v)", pollingPeriod)
 	}
 
-	db, ok := cfg[KeyDatabase]
-	if !ok || db == "" {
-		db = defaultDatabase
-	}
-
-	dbInt, err := strconv.Atoi(db)
-	if err != nil {
-		return Config{}, fmt.Errorf("invalid database passed, should be a valid int")
-	}
-
 	config := Config{
-		Host:          host,
 		RedisKey:      key,
-		Port:          port,
-		Database:      dbInt,
+		Host:          defaultHost,
+		Port:          defaultPort,
 		Password:      cfg[KeyPassword],
 		Username:      cfg[KeyUsername],
 		Mode:          ModePubSub,
 		PollingPeriod: pollingDuration,
 	}
+
+	if host := cfg[KeyHost]; host != "" {
+		config.Host = host
+	}
+
+	if port := cfg[KeyPort]; port != "" {
+		config.Port = port
+	}
+
+	if db := cfg[KeyDatabase]; db != "" {
+		dbInt, err := strconv.Atoi(db)
+		if err != nil {
+			return Config{}, fmt.Errorf("invalid database passed, should be a valid int")
```

Comment:
I think it would be better to replace the `fmt.Errorf` with the `errors.New`, cause we don't have any wrapping here.